### ISSUE TITLE
Average Precision and recall are -1

### ISSUE THREAD

**Issue by onosecond** [Tianxiaomo/pytorch-YOLOv4#164]
![image](https://user-images.githubusercontent.com/68383040/87672744-02d63700-c791-11ea-8057-6120c4dc7e64.png)

Not sure why this is happening. Are the bounding boxes supposed to be normalised?
Command i use for training -
`!python train.py -l 0.001 -g 0 -pretrained yolov4.conv.137.pth -classes 1 -dir "train_and_test" -train_label_path data/train.txt`

Here train_and_test contains the combined training and testing images.

The train.txt looks like this(val.txt is also the same format)- 
010dbcc8f.jpg 125,302,60,237,0 217,306,141,230,0 188,572,114,498,0 138,639,71,572,0 282,679,196,593,0
0114c88aa.jpg 601,304,514,217,0 702,133,586,17,0 395,979,322,906,0

Above data is of the format x1,y1,x2,y2,class_id
(img size is 1024)

Changes made in config file:
```
Cfg.use_darknet_cfg = False
Cfg.cfgfile = os.path.join(_BASE_DIR, 'cfg', 'yolov4.cfg')
Cfg.batch = 4
Cfg.subdivisions = 2
Cfg.width = 608
Cfg.height = 608
```
Here is my custom get_image_id function:
I think the problem might be here as the images id are alphanumeric so I just append all the image_ids in a list which is then converted to a dataframe to return an index(overkill i know)
 ```
  import pandas as pd
  df = pd.read_csv("./pytorch-YOLOv4/uniques.csv")
  no = df[df['names']==filename.split('.')[0]]['index'].to_list()[0]
  return int(no)
```
I have no idea what I am doing wrong. 
I have another question- Is it possible to train on 1024x1024 without resizing them directly? I understand that we need to provide the shape in the format 320+96*n, and hence I had provided 992(the closest to 1024) but I was getting an error something like
![image](https://user-images.githubusercontent.com/68383040/87673807-bb50aa80-c792-11ea-9c82-3cf829769aab.png)

I think I need to change the image_size in Yolo_loss Class as well for this to work? I am not sure.

**Comment by onosecond** [comment]
For anyone reading this- you have to change the format to xmin,ymin,w,h and not x1,y1,x2,y2 as mentioned [here](https://github.com/Tianxiaomo/pytorch-YOLOv4/blob/master/Use_yolov4_to_train_your_own_data.md). Please update it as it can be confusing to many
But i still dont know how to train on images other than 608. Please let me know how to do it

**Comment by Apucs** [comment]
@onosecond How are you plotting the result? can you help me with that?

**Comment by RicoSuaveGuapo** [comment]
> ![Screenshot from 2020-07-15 22-44-17](https://user-images.githubusercontent.com/68383040/87718636-c675fb80-c7cf-11ea-83c5-fdb0001a8d53.png)
> Not sure why this is happening. Are the bounding boxes supposed to be normalised?

I am facing the same issue. I change the line (around line 426) of dataset.py 
```py
boxes[..., 2:] = boxes[..., 2:] - boxes[..., :2]  # box width, box height
```
to
```
boxes[..., 2] = boxes[..., 2] - boxes[..., 0]
boxes[..., 3] = boxes[..., 1] - boxes[..., 3]
```
the AP and AR are not -1 anymore. **BUT! I am not certain about this modification !!**

The reason I change above is because that :
I saw the file (Use_yolov4_to_train_your_own_data) said that the `train.txt ` should be formatted as
```
image_path1 x1,y1,x2,y2,id x1,y1,x2,y2,id x1,y1,x2,y2,id ...
image_path2 x1,y1,x2,y2,id x1,y1,x2,y2,id x1,y1,x2,y2,id ...
```
where
- image_path : 图片名
- x1,y1 : 左上角坐标 (means top-left)
- x2,y2 : 右下角坐标 (means down-right)
- id : 物体类别

The coco format is [x,y,width,height] (x, y are top-left), therefore, if we want to change the format of `train.txt` into coco format, we should
- width = x2 - x1
- height = y1 - y2

This is modification I made above. 


Still, I am not sure about this correction, correct (or teach) me. Since even though the log shows AR and AP values, I still can not get any detection while inference @.@

**Comment by Jinhui-Qin** [comment]
Hi, have you solved this problem? My training AP and AR are -1
